src/mysql_k8s_helpers.py

```diff
@@ -785,17 +744,17 @@ def safe_stop_mysqld(self):
 
         def get_mysqld_pid(self):
             try:
-                process = self.container.exec(["pgrep", "-x", "mysqld"])
+                process = self.container.exec(["pgrep", "-x", MYSQLD_SAFE_SERVICE])
```

Comment:
This will return the pid of mysqld_safe and not mysqld, while function is get_mysqld_pid().
Is it fine? I believe on `RESTART` mysqld PID will be changed but mysqld_safe PID will remain.

Reply:
Additionally, does `mysqld_safe` fork `mysqld` process (being it parent process) or exec it (replacing the context)?